### ISSUE TITLE
NeonService push unit test failing

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#1095]
The `Test_Metrics.Push()` test failed during the unit test pass for v2.10.0 but I went ahead and published the release anyway, because this feature is probably not used yep.

The test spins for a while and it does appear to receive a metrics push from the emulated service but the test can't find the **test_counter** so it fails.

**Comment by jefflill** [comment]
So the service is in fact pushing metrics.  Here's what we're receiving:
```
# HELP process_start_time_seconds Start time of the process since unix epoch in seconds.
# TYPE process_start_time_seconds gauge
process_start_time_seconds 1606670438.0387683
# HELP dotnet_collection_count_total GC collection count
# TYPE dotnet_collection_count_total counter
dotnet_collection_count_total{generation="2"} 0
dotnet_collection_count_total{generation="1"} 1
dotnet_collection_count_total{generation="0"} 2
# HELP process_working_set_bytes Process working set
# TYPE process_working_set_bytes gauge
process_working_set_bytes 98680832
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total counter
process_cpu_seconds_total 1.34375
# HELP process_virtual_memory_bytes Virtual memory size in bytes.
# TYPE process_virtual_memory_bytes gauge
process_virtual_memory_bytes 2204157931520
# HELP neon_service_info Describes your service version.
# TYPE neon_service_info gauge
neon_service_info{version=""} 1
# HELP process_num_threads Total number of threads
# TYPE process_num_threads gauge
process_num_threads 22
# HELP process_private_memory_bytes Process private memory size
# TYPE process_private_memory_bytes gauge
process_private_memory_bytes 60121088
# HELP test_counter my test counter
# TYPE test_counter counter
test_counter 0
# HELP neon_log_events_total Number of logged events.
# TYPE neon_log_events_total counter
neon_log_events_total{level="info"} 1
# HELP dotnet_total_memory_bytes Total known allocated memory
# TYPE dotnet_total_memory_bytes gauge
dotnet_total_memory_bytes 6052184
# HELP process_open_handles Number of open handles
# TYPE process_open_handles gauge
process_open_handles 679
```
**test_counter** is in there but it hasn't been incremented past 0 which is why the test is failing.  The unit test also checks for log related counters which are not present, so the test will fail on that next.

**Comment by jefflill** [comment]
This ended up being a bad unit test.  This seemed like it was working before but we just got lucky with some timing.  I corrected a few problems:

- [ ] The unit test wasn't answering the HTTP push request from the service so the Prometheus client waited a really long time (maybe 2min) before timing out.  I fixed this by answering with 200 OK.

- [ ] The test also wasn't leaving the emulated Pushgateway listener open long enough for the metrics pusher to be able to push the last data before stopping.  This made it look like `MetricPusher.StopAsync()` was hanging (it will eventually time out).

The test is passing now.
